idaes/core/util/parameter_sweep.py

```diff
@@ -0,0 +1,728 @@
+    Config validator for ParameterSweepSpecifications
+
+    Args:
+        val: value to validate
+
+    Returns:
+        ParameterSweepSpecification
+
+    Raises:
+        ValueError if val is not a ParameterSweepSpecification
+    """
+    if isinstance(val, ParameterSweepSpecification):
+        return val
+    _log.error(
+        f"Input configuration {val} must be an instance of ParameterSweepSpecification."
+    )
+    raise ValueError(
+        "Input configuration must be an instance of ParameterSweepSpecification."
+    )
+
+
+CONFIG = ConfigDict()
+CONFIG.declare(
+    "build_model",
+    ConfigValue(doc="Callback method to construct initialized model for execution."),
+)
+CONFIG.declare(
+    "build_model_arguments",
+    ConfigValue(
+        domain=dict,
+        doc="Arguments to pass to build_model callback.",
+    ),
+)
+CONFIG.declare(
+    "run_model",
+    ConfigValue(
+        doc="Callback method to use when running model. If None, model is run with solver.solve()"
+    ),
+)
+CONFIG.declare(
+    "run_model_arguments",
+    ConfigValue(
+        domain=dict,
+        doc="Arguments to pass to run_model callback.",
+    ),
+)
+CONFIG.declare(
+    "collect_results",
+    ConfigValue(
+        doc="Callback method to use to collect results from model after execution."
+    ),
+)
+CONFIG.declare(
+    "collect_results_arguments",
+    ConfigValue(
+        domain=dict,
+        doc="Arguments to pass to collect_results callback.",
+    ),
+)
+CONFIG.declare(
+    "failure_recourse",
+    ConfigValue(
+        doc="Callback method to use in case of exception when solving a sample."
+    ),
+)
+CONFIG.declare(
+    "failure_recourse_arguments",
+    ConfigValue(
+        domain=dict,
+        doc="Arguments to pass to failure_recourse callback.",
+    ),
+)
+CONFIG.declare(
+    "input_specification",
+    ConfigValue(
+        domain=is_psweepspec,
+        doc="ParameterSweepSpecification object defining inputs to be sampled",
+    ),
+)
+CONFIG.declare(
+    "solver",
+    ConfigValue(
+        default=None,
+        domain=str,
+        doc="Name of solver to use (recognised by Pyomo SolverFactory)",
+    ),
+)
+CONFIG.declare(
+    "solver_options",
+    ConfigValue(
+        default=None,
+        domain=dict,
+        doc="Dict of options to pass to solver",
+    ),
+)
+
+
+@document_kwargs_from_configdict(CONFIG)
+class ParameterSweepBase:
+    """
+    IDAES base class for defining parameter sweep runners.
+
+    Thus base class defines a basic API for setting up parameter sweep studies
+    and allows the end-user to provide instructions for how to set up and run
+    the model to be studied using a set of callbacks.
+    """
+
+    def __init__(self, **kwargs):
+        self.config = CONFIG(kwargs)
+        self._results = OrderedDict()
+
+    @property
+    def results(self):
+        """
+        Returns OrderedDict containing the results from the parameter sweep.
+        """
+        return self._results
```

Comment:
Not sure on that - I don't see any reason why it could not be, but I am also not sure that it needs to be either. The reason the `ParameterSweepSpecification` uses a `DataFrame` is because Pysmo returns a `numpy.array` (which I then turned into a `DataFrame` so that the input names were explicitly associated with the data to avoid future issues).

Reply:
For now I think I want to leave things as they are. For one, I think it is best to keep the samples separate from the results (we've tried different things at different time and for now I prefer this).